sdk/logical/response.go

```diff
@@ -177,3 +179,25 @@ func RespondWithStatusCode(resp *Response, req *Request, code int) (*Response, e
 
 	return ret, nil
 }
+
+type ResponseWriter struct {
```

Comment:
Maybe call this something else as we also have forwarding.RPCResponseWriter. They're different enough that they likely can't be merged, but that's kind of my point here, which is this is a custom writer for something very specific so the generic name feels a bit odd.